### PR TITLE
docs(dynamic-plugins): add frontend docs

### PR DESCRIPTION
**Pull request by tumido**
## Description

Documentation to https://github.com/janus-idp/backstage-showcase/pull/700

## Which issue(s) does this PR fix

Part of https://github.com/janus-idp/backstage-showcase/issues/442

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

### CODE REVIEW

File: showcase-docs/dynamic-plugins.md [hunk 1]
```diff
@@ -33,7 +35,7 @@ So there are some changes to be made to the plugin code, in order to make it com
 1. The plugin must:
 
 - import the `@backstage/backend-plugin-manager` package, as an alias to `janus-idp/backend-plugin-manager@0.0.5-janus.0` package,
-- add the `@janus-idp/cli` dependency to use the `@janus-idp` fork of the backstage CLI that supports a new, required `export-dynamic-plugin` command,
+- add the `@janus-idp/cli` dependency to use the `@janus-idp`,
```
Comment (davidfestal) [inline showcase-docs/dynamic-plugins.md:38]:
```suggestion
- add the `@janus-idp/cli` dependency, which provides a new, required, `export-dynamic-plugin` command.
```

File: showcase-docs/dynamic-plugins.md [hunk 2]
```diff
@@ -313,3 +394,193 @@ upstream:
...
+      dynamicRoutes: # exposes full routes
+        - path: /my-plugin # unique path in the app, can override `/`
+          module: CustomModule # optional, same as key in `scalprum.exposedModules` key in plugin's `package.json`
+          importName: FooPluginPage # optional, actual component name that should be rendered
+          menuItem: # optional, allows you to populate main sidebar navigation
+            icon: Storage # MUI4 icon to render in the sidebar
+            text: Foo Plugin Page # menu item text
+```
+
+Each plugin can expose multiple routes and each route is required to define it's `path` and `importName` (if it differs from the default export).
```
Comment (nickboldt) [inline showcase-docs/dynamic-plugins.md:474]:
```suggestion
Each plugin can expose multiple routes and each route is required to define its `path` and `importName` (if it differs from the default export).
```

File: showcase-docs/dynamic-plugins.md [hunk 3]
```diff
@@ -313,3 +394,193 @@ upstream:
...
+          module: CustomModule # optional, same as key in `scalprum.exposedModules` key in plugin's `package.json`
+          importName: FooPluginPage # optional, actual component name that should be rendered
+          menuItem: # optional, allows you to populate main sidebar navigation
+            icon: Storage # MUI4 icon to render in the sidebar
+            text: Foo Plugin Page # menu item text
+```
+
+Each plugin can expose multiple routes and each route is required to define its `path` and `importName` (if it differs from the default export).
+
+- `path` - Unique path in the app, cannot override existing routes with single exception: the `/` home route. Main home page can be replaced via dynamic plugins mechanism.
```
Comment (nickboldt) [inline showcase-docs/dynamic-plugins.md:476]:
```suggestion
- `path` - Unique path in the app. Cannot override existing routes with the exception of the `/` home route: the main home page can be replaced via the dynamic plugins mechanism.
```

File: showcase-docs/dynamic-plugins.md [hunk 4]
```diff
@@ -313,3 +394,193 @@ upstream:
...
+        ...
+      </FlatRoutes>
+    </Root>
+  </AppRouter>
+...
+```
+
+This change is usually coupled with extension to the main sidebar navigation by editing `packages/app/src/components/Root/Root.tsx`.
+
+In dynamic plugins this mechanism has changed and users are no longer allowed to edit TSX files - instead they declare their desire to expose additional routes within dynamic plugin configuration:
```
Comment (nickboldt) [inline showcase-docs/dynamic-plugins.md:458]:
```suggestion
In dynamic plugins this mechanism has changed and users are no longer allowed to edit `.tsx` files. Instead they declare their desire to expose additional routes within dynamic plugin configuration:
```

File: showcase-docs/dynamic-plugins.md [hunk 5]
```diff
@@ -313,3 +394,193 @@ upstream:
...
+        {/* Additional routes defined by user */}
+        <Route path="/my-plugin" element={<FooPluginPage />} />
+        ...
+      </FlatRoutes>
+    </Root>
+  </AppRouter>
+...
+```
+
+This change is usually coupled with extension to the main sidebar navigation by editing `packages/app/src/components/Root/Root.tsx`.
```
Comment (nickboldt) [inline showcase-docs/dynamic-plugins.md:456]:
```suggestion
This change is usually coupled with an extension to the main sidebar navigation, achieved by editing `packages/app/src/components/Root/Root.tsx`.
```

File: showcase-docs/dynamic-plugins.md [hunk 6]
```diff
@@ -313,3 +394,193 @@ upstream:
...
+# app-config.yaml
+dynamicPlugins:
+  frontend:
+    <package_name>: # same as `scalprum.name` key in plugin's `package.json`
+      dynamicRoutes: # exposes full routes
+        - path: /my-plugin # unique path in the app, can override `/`
+          module: CustomModule # optional, same as key in `scalprum.exposedModules` key in plugin's `package.json`
+          importName: FooPluginPage # optional, actual component name that should be rendered
+          menuItem: # optional, allows you to populate main sidebar navigation
+            icon: Storage # MUI4 icon to render in the sidebar
```
Comment (davidfestal) [inline showcase-docs/dynamic-plugins.md:470]: Is there the ability to use a provided icon (not MUI-standard) ?

File: showcase-docs/dynamic-plugins.md [hunk 7]
```diff
@@ -313,3 +394,193 @@ upstream:
...
+  dynamic:
+    plugins:
+      - package: <packageName>
+        pluginConfig:
+          # app-config.yaml fragment
+```
+
+### Frontend layout configuration
+
+Compared to the backend plugins, where their mount points are defined in code and consumed by the backend plugin manager, frontend plugins require additional configuration in the `app-config.yaml`. A pluggin missing this configuration is not loaded into the application and is not displayed.
```
Comment (nickboldt) [inline showcase-docs/dynamic-plugins.md:414]:
```suggestion
Compared to the backend plugins, where mount points are defined in code and consumed by the backend plugin manager, frontend plugins require additional configuration in the `app-config.yaml`. A plugin missing this configuration will not be loaded into the application and will not be displayed.
```

File: showcase-docs/dynamic-plugins.md [hunk 8]
```diff
@@ -313,3 +394,193 @@ upstream:
...
+
+Similarly to what you know from traditional Backstage instances, there are 3 types of functionality a dynamic frontend plugin can offer:
```
Comment (nickboldt) [inline showcase-docs/dynamic-plugins.md:416]:
```suggestion
Similarly to traditional Backstage instances, there are 3 types of functionality a dynamic frontend plugin can offer:
```

File: showcase-docs/dynamic-plugins.md [hunk 9]
```diff
@@ -313,3 +394,193 @@ upstream:
...
+
+- Full new page by declaring a completely new route in the app
```
Comment (nickboldt) [inline showcase-docs/dynamic-plugins.md:418]:
```suggestion
- Full new page that declares a completely new route in the app
```

File: showcase-docs/dynamic-plugins.md [hunk 10]
```diff
@@ -313,3 +394,193 @@ upstream:
...
+- Extend existing page via router `bind`ings
```
Comment (nickboldt) [inline showcase-docs/dynamic-plugins.md:419]:
```suggestion
- Extension to existing page via router `bind`ings
```

File: showcase-docs/dynamic-plugins.md [hunk 11]
```diff
@@ -313,3 +394,193 @@ upstream:
...
+- Using mount points within the application
```
Comment (nickboldt) [inline showcase-docs/dynamic-plugins.md:420]:
```suggestion
- Use of mount points within the application
```

**Review by davidfestal — APPROVED** [review]
Approved with 1 question and 1 suggestion

File: showcase-docs/dynamic-plugins.md [hunk 12]
```diff
@@ -313,3 +394,193 @@ upstream:
...
+
+- `path` - Unique path in the app, cannot override existing routes with single exception: the `/` home route. Main home page can be replaced via dynamic plugins mechanism.
+- `module` - Optional. Since dynamic plugins can expose multiple distinct modules, you may need to specify which set of assets you want to access within the plugin. If not provided, the default module named `PluginRoot` is used. This is the same as the key in `scalprum.exposedModules` key in plugin's `package.json`.
+- `importName` - Optional. The actual component name that should be rendered as a standalone page. If not specified the `default` export is used.
+- `menuItem` - This property allows users to extend the main sidebar navigation and point to their new route. It accepts `text` and `icon` properties. `icon` is a Material UI 4 icon name.
+
+#### Bind to existing plugins
+
+Another extension option Backstage offers is extension via [binding to the external routes](https://backstage.io/docs/plugins/composability/#binding-external-routes-in-the-app) of existing plugins. This is traditionally done via the `bindRoutes` interface as:
```
Comment (nickboldt) [inline showcase-docs/dynamic-plugins.md:483]:
```suggestion
Another extension option available to Backstage is to [bind to the external routes](https://backstage.io/docs/plugins/composability/#binding-external-routes-in-the-app) of existing plugins. This is traditionally done via the `bindRoutes` interface as:
```

File: showcase-docs/dynamic-plugins.md [hunk 13]
```diff
@@ -313,3 +394,193 @@ upstream:
...
+dynamicPlugins:
+  frontend:
+    <package_name>: # same as `scalprum.name` key in plugin's `package.json`
+      routeBindings:
+        - bindTarget: 'barPlugin.externalRoutes' # One of the supported bind targets
+          bindMap: # Map of bindings, same as the `bind` function options argument in the example above
+            headerLink: 'fooPlugin.routes.root'
+```
+
+We currently support following bind targets:
```
Comment (nickboldt) [inline showcase-docs/dynamic-plugins.md:508]:
```suggestion
These are the available bind targets:
```

File: showcase-docs/dynamic-plugins.md [hunk 14]
```diff
@@ -313,3 +394,193 @@ upstream:
...
+
+#### Using mount points
+
+This section aims to allow users dynamic extension of [Catalog Components](https://backstage.io/docs/plugins/composability/#catalog-components), but can be used to extend additional views in the future as well.
+
+Mount points are defined identifiers available across the applications. These points specifically allow users to extend existing pages with additional content.
+
+We currently recognize following mount points:
+
+| Mount point                  | Description                         | Content with all plugins disabled                              |
```
Comment (nickboldt) [inline showcase-docs/dynamic-plugins.md:523]:
```suggestion
| Mount point                  | Description                         | Visible even when no plugins are enabled                              |
```

File: showcase-docs/dynamic-plugins.md [hunk 15]
```diff
@@ -313,3 +394,193 @@ upstream:
...
+| `entity.page.lighthouse`     | Catalog entity "Lighthouse" tab     | NO                                                             |
+| `entity.page.api`            | Catalog entity "API" tab            | YES for entity of `kind: Component` and `spec.type: 'service'` |
+| `entity.page.dependencies`   | Catalog entity "Dependencies" tab   | YES for entity of `kind: Component`                            |
+| `entity.page.docs`           | Catalog entity "Documentation" tab  | YES for entity that satisfies `isTechDocsAvailable`            |
+| `entity.page.definition`     | Catalog entity "Definitions" tab    | YES for entity of `kind: Api`                                  |
+| `entity.page.diagram`        | Catalog entity "Diagram" tab        | YES for entity of `kind: System`                               |
+
+Note: Mount points within Catalog aka `entity.page.*` are rendered as tabs. They become visible only if at least one plugin contributes to them or they can render static content (see column 3 in previous table).
+
+Each mount point exists in 2 variants that complement each other :
```
Comment (nickboldt) [inline showcase-docs/dynamic-plugins.md:544]:
```suggestion
Each mount point has 2 complementary variations:
```

File: showcase-docs/dynamic-plugins.md [hunk 16]
```diff
@@ -313,3 +394,193 @@ upstream:
...
+            props: {} # React props to pass to the component
+```
+
+Each mount point supports additional configuration:
+
+- `layout` - Used only in `*/cards` type which renders visible content. Allows you to pass MUI sx properties to the component. This is useful when you want to control the layout of the component. `entity.page.*` mount points are rendered as CSS grid, so SX property allows you to control the grid layout and exact positioning of the rendered component.
+- `props` - React props passed to the component. This is useful when you want to pass some additional data to the component.
+- `if` - Used only in `*/cards` type which renders visible content. This is passed to `<EntitySwitch.Case if={<here>}`.
+
+  We support following conditions:
```
Comment (nickboldt) [inline showcase-docs/dynamic-plugins.md:575]:
```suggestion
  The following conditions are available:
```

**Review by nickboldt — APPROVED** [review]
skimmed and did some language review; looks OK to me. 

probably more language tweaks could be done but until @hmanwani-rh can get 👀 on it, I think it's probably OK to merge as is and be cleaned up later if/when we're downstreaming this into one of the RHDH user / admin guides

**Pull request merged**